src/repository/Repository.ts

```diff
@@ -49,7 +50,9 @@ export class Repository<M extends Model = Model> {
   initialize(model?: ModelConstructor<M>): this {
     // If there's a model passed in, just use that and return immediately.
     if (model) {
-      this.model = model.newRawInstance().$setDatabase(this.database)
+      this.model = model.newRawInstance()
+
+      this.$setInversePolymorphicDatabase(this.model)
```

Comment:
Injecting database into Inverse Polymorphic Relations so that related data can be correctly attached and retrieved